### PR TITLE
[BEAM-7027] Use same method to find a new local available port in IO tests

### PR DESCRIPTION
**Pull request by iemejia**
R: @aromanenko-dev

### CODE REVIEW

**Comment by iemejia** [comment]
Run Java PreCommit

**Comment by iemejia** [comment]
Run Java PreCommit

File: sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java [hunk 1]
```diff
@@ -134,6 +133,15 @@ public static void startCassandra() throws Exception {
     cassandraPort = cluster.getConfiguration().getProtocolOptions().getPort();
     session = CassandraIOTest.cluster.newSession();
 
+    createData();
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java]: ok

File: sdks/java/io/clickhouse/src/test/java/org/apache/beam/sdk/io/clickhouse/AtomicInsertTest.java [hunk 2]
```diff
@@ -39,13 +38,11 @@
 import org.joda.time.Duration;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 /** Tests for atomic/idempotent inserts for {@link ClickHouseIO}. */
 @RunWith(JUnit4.class)
-@Category(NeedsRunner.class)
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/clickhouse/src/test/java/org/apache/beam/sdk/io/clickhouse/AtomicInsertTest.java]: IO tests are not filtered for test execution, the main goal of categorization was to do such filtering e.g. to run ValidatesRunner tests, so it makes less sense in IO. @jkff did a PR long time ago that removed all of them so I follow his advice.

File: sdks/java/io/elasticsearch-tests/elasticsearch-tests-5/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOIT.java [hunk 3]
```diff
@@ -49,6 +51,7 @@
  * <p>It is likely that you will need to configure <code>thread_pool.bulk.queue_size: 250</code> (or
  * higher) in the backend Elasticsearch server for this test to run.
  */
+@RunWith(JUnit4.class)
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/elasticsearch-tests/elasticsearch-tests-5/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOIT.java]: You need this in general to tell JUnit to use a defined runner, I put it in the IT to be consistent with the other ITs that already had it, and be sure that some weird default (e.g. JUnit 3 in some module won't break reproducibility).

File: sdks/java/io/elasticsearch-tests/elasticsearch-tests-6/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOIT.java [hunk 4]
```diff
@@ -49,6 +51,7 @@
  * <p>It is likely that you will need to configure <code>thread_pool.bulk.queue_size: 250</code> (or
  * higher) in the backend Elasticsearch server for this test to run.
  */
+@RunWith(JUnit4.class)
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/elasticsearch-tests/elasticsearch-tests-6/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOIT.java]: answered above

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java [hunk 5]
```diff
@@ -223,7 +220,6 @@ public void testRuntimeValueProviderTopic() {
   }
 
   @Test
-  @Category({ValidatesRunner.class, UsesUnboundedPCollections.class})
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java]: answered above

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java [hunk 6]
```diff
@@ -264,7 +260,6 @@ public void testWriteDisplayData() {
   }
 
   @Test
-  @Category(ValidatesRunner.class)
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java]: answered above

File: sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java [hunk 7]
```diff
@@ -110,8 +109,8 @@
   private static CassandraShutDownHook shutdownHook;
 
   @BeforeClass
-  public static void startCassandra() throws Exception {
-    jmxPort = getFreeLocalPort();
+  public static void beforeClass() throws Exception {
+    jmxPort = NetworkTestHelper.getAvailableLocalPort();
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java]: Yes, I thought about this case too, will try to tackle it. I will ping you when done

File: sdks/java/io/clickhouse/src/test/java/org/apache/beam/sdk/io/clickhouse/ClickHouseIOTest.java [hunk 8]
```diff
@@ -38,13 +37,11 @@
 import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 /** Tests for {@link ClickHouseIO}. */
 @RunWith(JUnit4.class)
-@Category(NeedsRunner.class)
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/clickhouse/src/test/java/org/apache/beam/sdk/io/clickhouse/ClickHouseIOTest.java]: The same answer as above

**Comment by iemejia** [comment]
PR Updated, PTAL @aromanenko-dev

**Comment by aromanenko-dev** [comment]
LGTM
Please, self-merge it once the checks will be green.

**Pull request merged**